### PR TITLE
feat: Added Confidence Score for TextBlock/Line/Element

### PR DESCRIPTION
**Pull request by bensonarafat**
@fbernaly based on #448 issue to add confidence score for TextBlock/Line/Element on `google_mlkit_text_recognition`. 
Just added the `confidence score` and `angle`. However, TextBlock doesn't give these two on the [ML kit doc](https://developers.google.com/android/reference/com/google/mlkit/vision/text/Text.TextBlock)  
Also `google_mlkit_language_id` fix the missing id for `identifyLanguage`

### CODE REVIEW

**Comment by EduardoDiNizo** [comment]
@bensonarafat , Is it necessary to include the mapping of confidence and angle fields in the Java and IOS implementation as well?

example: packages/google_mlkit_text_recognition/android/src/main/java/com/google_mlkit_text_recognition/TextRecognizer.java
```
Line 95
addData(blockData,
               block.getText(),
               block.getBoundingBox(),
               block.getCornerPoints(),
               block.getRecognizedLanguage(),
               block.getConfidence(),
               block.getAngle());

LIne 105               
addData(lineData,
               line.getText(),
               line.getBoundingBox(),
               line.getCornerPoints(),
               line.getRecognizedLanguage()
               line.getConfidence(),
               line.getAngle());

Line 115            
addData(elementData,
               element.getText(),
               element.getBoundingBox(),
               element.getCornerPoints(),
               element.getRecognizedLanguage(),
               element.getConfidence(),
               element.getAngle());        
```
I didn't implement the solution but I saw that these fields are not mapped in the native implementations either.
It would be interesting to include these two fields in the example available in the project as well.

Thank you for creating the pull request.

**Comment by bensonarafat** [comment]
@EduardoDiNizo
Oh yeah, thanks

**Comment by bensonarafat** [comment]
@EduardoDiNizo 
Text block do not return confidence score. 
You can verify it [here](https://developers.google.com/android/reference/com/google/mlkit/vision/text/Text.TextBlock)

**Comment by EduardoDiNizo** [comment]
@bensonarafat my bad.
Confidence score is present in Line, Element and Symbol I think.

**Comment by fbernaly** [comment]
@bensonarafat: I was about to suggest the same as @EduardoDiNizo, you need to update the ios and android part as well to get those new properties from the native API, also update your working branch with the latest code from our master.
Thanks for the contribution.

**Comment by bensonarafat** [comment]
> @bensonarafat my bad.
> Confidence score is present in Line, Element and Symbol I think.

yeap

**Comment by bensonarafat** [comment]
> @bensonarafat: I was about to suggest the same as @EduardoDiNizo, you need to update the ios and android part as well to get those new properties from the native API, also update your working branch with the latest code from our master.

yeah, thanks

**Comment by bensonarafat** [comment]
@fbernaly I just sent an update, you can make a review and merge now 
thanks

File: packages/google_mlkit_text_recognition/android/src/main/java/com/google_mlkit_text_recognition/TextRecognizer.java [hunk 1]
```diff
@@ -145,8 +154,11 @@ private void addData(Map<String, Object> addTo,
         addTo.put("rect", getBoundingPoints(rect));
         addTo.put("recognizedLanguages", recognizedLanguages);
         addTo.put("text", text);
+        addTo.put("confidence", confidence);
+        addTo.put("angle", angle);
     }
 
+
```
Comment (fbernaly) [inline packages/google_mlkit_text_recognition/android/src/main/java/com/google_mlkit_text_recognition/TextRecognizer.java:161]: remove extra line

File: packages/google_mlkit_text_recognition/android/src/main/java/com/google_mlkit_text_recognition/TextRecognizer.java [hunk 2]
```diff
@@ -156,6 +168,7 @@ private void addPoints(Point[] cornerPoints, List<Map<String, Integer>> points)
         }
     }
 
+
```
Comment (fbernaly) [inline packages/google_mlkit_text_recognition/android/src/main/java/com/google_mlkit_text_recognition/TextRecognizer.java:171]: remove extra line

File: packages/google_mlkit_text_recognition/ios/Classes/GoogleMlKitTextRecognitionPlugin.m [hunk 3]
```diff
@@ -157,7 +163,9 @@ - (void)addData:(NSMutableDictionary *)addTo
    cornerPoints:(NSArray<NSValue *> *)cornerPoints
           frame:(CGRect)frame
       languages:(NSArray<MLKTextRecognizedLanguage *> *)languages
-           text:(NSString *)text {
+           text:(NSString *)text
```
Comment (fbernaly) [inline packages/google_mlkit_text_recognition/ios/Classes/GoogleMlKitTextRecognitionPlugin.m:166]: you are missing to add `confidence` and `angle` to the dictionary inside this method. Look at the end of that method and add it in `addEntriesFromDictionary`

**Comment by fbernaly** [comment]
@bensonarafat : I added a couple of comments

File: packages/google_mlkit_language_id/lib/src/language_identifier.dart [hunk 4]
```diff
@@ -23,9 +23,10 @@ class LanguageIdentifier {
   /// If no language could be determined then [undeterminedLanguageCode] is returned.
   /// More information: https://developers.google.com/ml-kit/language/identification
   Future<String> identifyLanguage(String text) async {
-    final result = await _channel.invokeMethod(
-        'nlp#startLanguageIdentifier', <String, dynamic>{
+    final result = await _channel
+        .invokeMethod('nlp#startLanguageIdentifier', <String, dynamic>{
       'text': text,
+      'id': id,
```
Comment (fbernaly) [inline packages/google_mlkit_language_id/lib/src/language_identifier.dart:29]: could you please add `id` at the end of the map, just as it is in `identifyPossibleLanguages`, for the sake of having same estructure... cosmetic and readability

**Comment by bensonarafat** [comment]
@fbernaly thanks for the review.. just updated it

**Review by fbernaly — APPROVED** [review]

**Comment by EduardoDiNizo** [comment]
@fbernaly , @bensonarafat . Thank you very much for your attention and agility in this task. I will be the first to test the release! :)

**Comment by EduardoDiNizo** [comment]
Can I close this issue?

**Comment by fbernaly** [comment]
I will release by the end of the week, but in the meantime you can do something like this:

```
google_mlkit_language_id:
    git:
      url: git@github.com:flutter-ml/google_ml_kit_flutter.git
      ref: master
      path: packages/google_mlkit_language_id
```

**Comment by EduardoDiNizo** [comment]
text 512433 confidence 0.9342448115348816
text Qty confidence 0.4778645932674408
text 1025 confidence 0.689453125
text 12/31/2025 confidence 0.7777343988418579

It's working! Thanks

**Comment by fbernaly** [comment]
@bensonarafat: did you check the iOS API? confidence and angle are not supported in iOS: https://developers.google.com/ml-kit/reference/ios/mlkittextrecognitioncommon/api/reference/Classes/MLKTextLine
I need to update the release.
Someone reported: https://github.com/flutter-ml/google_ml_kit_flutter/issues/458

**Comment by bensonarafat** [comment]
@fbernaly  I notice that too. it is not supported for IOS. My bad.. 
I could send another PR setting IOS to `NULL` 
Or which way is better?

**Comment by EduardoDiNizo** [comment]
Hunn, I didn't expect that either.

It's weird that Google doesn't implement the same features on both platforms.

Is it possible to maintain the Android implementation?

Will Google make this information available in the future?

**Comment by fbernaly** [comment]
I dont know Google's ETA, in the meantime I will make those properties optional, wait for a new PR

**Comment by fbernaly** [comment]
> @fbernaly I notice that too. it is not supported for IOS. My bad..
> I could send another PR right away setting IOS to `NULL`
> Or which way is better?

I am working on it

**Pull request merged**